### PR TITLE
Enhanced SEO Content Selection and Aggregation

### PR DESCRIPTION
**Pull request by neutral-Stage**
Enhanced SEO Content Selection and Aggregation

### The Problem
The plugin starts by attempting to select the `content` of an HTML element with a given selector (`contentSelector`) using `querySelector`. However, it was limited to selecting content from a single HTML element. Sometimes it grabs some unnecessary elements from the root that we might not want to detect, such as author image alternative text, and a related post article image alt text in a blog post. This becomes a common problem if a developer has structured their content in a way where not all the HTML elements they want analyzed for SEO are not located in the Rich text editor for the content body of a blog article, for example. 

### The Enhanced Solution:

1. Ability to select Multiple Elements: Provide a way for the developer can target multiple elements on a page to be analyzed by the SEO plugin. These elements are identified by the `[data-content="seo"]` attribute.

2. Aggregate Content: Instead of selecting content from a single element, we want to aggregate the content from all the elements with the `[data-content="seo"]` attribute into a single container (`mainElement`).

3. Fallback Mechanism: If there are no elements with the `[data-content="seo"]` attribute on the page, the plugin will have a fallback mechanism to retrieve content using the original `contentSelector`.

By making these changes, we are adapting the code to handle more diverse scenarios where there might be multiple content instances with the `[data-content="seo"]` attribute on the page. This enhanced solution ensures we can effectively aggregate and handle this content, providing a more robust and flexible solution to your problem.

In summary, the problem we are addressing is the need to target multiple HTML elements that need to be analyzed for SEO by the plugin and ensure that our code can gracefully handle scenarios with or without these elements.

Example of selecting multiple contents of a blog post article:
```
<div>
 <h2 className=’blog-subtitle’ data-content="seo">
          		{post?.excerpt}
 	 </h2>
<!--Start of the element we want to avoid -->
<ArticleSideBarSticky
            author={post?.author}
            publishedAt={post?.publishedAt}
            categories={post?.categories}
          />
<!--End of the element we want to avoid -->
          <Richtext postBody={post?.body} data-content="seo" />
 </div>
```

### CODE REVIEW

**Comment by LiahMartens** [comment]
@neutral-Stage I'm not sure if it makes sense to implement specific use cases like this. You can use the `prepare` function instead to achieve this behavior.

ie:

```
S.view
  .component(SeoToolsPane)
  .options({
    fetch: false,
    resolveProductionUrl: (doc) =>
      new URL(`https://sanity.io/${doc?.slug?.current}`),
    select: (doc) => ({
      focus_keyword: doc.focus_keyword ?? "",
      seo_title: doc.seo_title ?? "",
      meta_description: doc.meta_description ?? "",
      focus_synonyms: doc.focus_synonyms ?? [],
    }),
    prepare: async (doc) => {
      const response = await fetch(``);
      const html = await response.text();
      const parser = new DOMParser()
      const doc = parser.parseFromString(html, 'text/html');
      const nodeList = doc.querySelectorAll(`[data-content='seo']`)
      
      const createElement = doc.createElement('main')
      await Promise.all(
        Array.from(nodeList).map((n) => createElement.appendChild(n))
      );
      const contentHtml = doc.querySelector(contentSelector)?.innerHTML || ''
      const mainElement = createElement?.innerHTML
      const content = nodeList?.length ? mainElement : contentHtml

      return {
        title: doc.seo_title ?? '',
        description: doc.meta_description ?? '',
        locale: doc.__i18n_lang ?? 'en-US',
        content,
      }
    },
  })
  .title("SEO");
```
